enterprise/ha/src/test/java/org/neo4j/kernel/ha/lock/ClusterLocksIT.java

```diff
@@ -83,49 +84,26 @@ public void lockCleanupOnModeSwitch() throws Throwable
     @Test
     public void aPendingMemberShouldBeAbleToServeReads() throws Throwable
     {
+        // given
         createNodeOnMaster( testLabel, cluster.getMaster() );
         cluster.sync();
+
         HighlyAvailableGraphDatabase slave = cluster.getAnySlave();
+        cluster.fail( slave, ClusterManager.NetworkFlag.values() );
+        cluster.await( instanceEvicted( slave ) );
 
-        AtomicBoolean run = new AtomicBoolean( true );
-        AtomicReference<Throwable> ref = new AtomicReference<>(  );
+        assertEquals( PENDING, slave.getInstanceState() );
```

Comment:
Ok the code / test framework is a bit weird here. It seems instanceEvicted means that the particular member itself thinks that it has the role of UNKNOWN. I thought it would be how the remaining members view the state of the slave and the transitions to evicted and pending would be asynchronous.